### PR TITLE
Continuation of support for ES3/WebGL2

### PR DESCRIPTION
**Pull request by chemaguerra**
This is in response to:
- https://github.com/raysan5/raylib/discussions/2447
- https://github.com/raysan5/raylib/commit/e497603678823dc93f01dd9686819a92d780ad67

I continued the work that @raysan5 started for OpenGL ES 3.0 context creation:

1- Fixed `#define GRAPHICS_API_OPENGL_33`.
2- Adapted `rlActiveDrawBuffers()` which now can be used to successfully initialize MRTs in WebGL2.
3- Did a (dirty) update to `RLGL.ExtSupported.*` since some ES 2.0 feature extensions can be taken for granted in ES 3.0.
4- Adapted `rlGetGlTextureFormats()` to support floating-point textures in ES 3.0.

5- I also edited the Makefile so raylib builds for ES3 (maybe exclude this on PR merge...).

The changes I am PR'ing have been tested in a couple of experimental projects I am conducting at the moment, in a bunch of web browsers across multiple devices (Windows, MacOS, Android, iPad, iPhone).

(3) needs some work to correctly detect the capabilities of the device. i.e., whether the device supports linear filtering for floating-point textures (many devices don't).

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -245,7 +245,8 @@ ifeq ($(PLATFORM),PLATFORM_DRM)
 endif
 ifeq ($(PLATFORM),PLATFORM_WEB)
     # On HTML5 OpenGL ES 2.0 is used, emscripten translates it to WebGL 1.0
-    GRAPHICS = GRAPHICS_API_OPENGL_ES2
+    #GRAPHICS = GRAPHICS_API_OPENGL_ES2
```
Comment (raysan5) [inline src/Makefile:248]: I prefer to keep `GRAPHICS_API_OPENGL_ES2` as the default, instead of `GRAPHICS_API_OPENGL_ES3`

File: src/Makefile [hunk 2]
```diff
@@ -344,7 +345,8 @@ endif
 
 ifeq ($(RAYLIB_BUILD_MODE),RELEASE)
     ifeq ($(PLATFORM),PLATFORM_WEB)
-        CFLAGS += -Os
+        #CFLAGS += -Os
```
Comment (raysan5) [inline src/Makefile:348]: I prefer to keep the `-Os` on web instead of `-O3`, from [emscripten documentation](https://emscripten.org/docs/optimizing/Optimizing-Code.html#how-to-optimize-code):

> Building with -O3 or -Os can produce an ever better build than -O2, and are worth considering for release builds. -O3 builds are even more optimized than -O2, but at the cost of significantly longer compilation time and potentially larger code size. -Os is similar in increasing compile times, but focuses on reducing code size while doing additional optimization.

File: src/raylib.h [hunk 3]
```diff
@@ -1149,6 +1149,7 @@ RLAPI int GetTouchX(void);                                    // Get touch posit
 RLAPI int GetTouchY(void);                                    // Get touch position Y for touch point 0 (relative to screen size)
 RLAPI Vector2 GetTouchPosition(int index);                    // Get touch position XY for a touch point index (relative to screen size)
 RLAPI int GetTouchPointId(int index);                         // Get touch point identifier for given index
+RLAPI bool GetTouchPointState(int index);                     // Patch for touch events in web browsers
```
Comment (raysan5) [inline src/raylib.h:1152]: This addition does not seem to belong to this PR.

**Comment by raysan5** [comment]
@chemaguerra Thank you very much for the addition and the testing! Just added a couple of comments.

File: src/Makefile (hunk 1 above)
Comment (chemaguerra) [reply src/Makefile:248]: Regarding your 3 comments: my bad! They were not intended to be part of the PR. Those are changes I made afterwards that I require for my project but are not meant to be merged into raylib. Switching by default to ES3 would break existing projects indeed.

File: src/Makefile (hunk 2 above)
Comment (chemaguerra) [reply src/Makefile:348]: Same. In my project -O3 runs significantly faster than -Os and the extra build time/size is a price I am willing to pay. But this change wasn't meant to be part of the PR.

File: src/raylib.h (hunk 3 above)
Comment (chemaguerra) [reply src/raylib.h:1152]: Same. Changes I made afterwards not meant to be part of this PR. Actually, this is in relation to this issue: https://github.com/raysan5/raylib/issues/3100

**Comment by chemaguerra** [comment]
I made some amends following your comments.

**Comment by raysan5** [comment]
@chemaguerra thanks for the review!
